src/utils/StreamMessageValidator.ts

```diff
@@ -140,18 +149,23 @@ export default class StreamMessageValidator {
         const stream = await this.getStream(streamMessage.getStreamId())
 
         // Checks against stream metadata
-        if (stream.requireSignedData && !streamMessage.signature) {
-            throw new ValidationError(`This stream requires data to be signed. Message: ${streamMessage.serialize()}`)
+        if ((stream.requireSignedData || this.requireSignedData) && !streamMessage.signature) {
+            throw new ValidationError(`Stream data is required to be signed. Message: ${streamMessage.serialize()}`)
         }
-        if (stream.requireEncryptedData && streamMessage.encryptionType === StreamMessage.ENCRYPTION_TYPES.NONE) {
-            throw new ValidationError(`This stream requires data to be encrypted. Message: ${streamMessage.serialize()}`)
+
+        if (
+            (stream.requireEncryptedData || this.requireEncryptedData) 
```

Comment:
Hmm.. I'm thinking this one isn't correct because the flag doesn't have any effect at all here. (This check is only done when `this.requireEncryptedData == true` as was the case before.) However, I'm unsure what the intention of the ticket here is? My guess is that we can't solely rely on `stream.requireEncryptedData` but instead need to also check whether the stream in question is private.